integration/all/integration_test.go

```diff
@@ -0,0 +1,312 @@
+// Copyright 2019-2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"sync/atomic"
+	"syscall"
+	"testing"
+
+	. "github.com/inspektor-gadget/inspektor-gadget/integration"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+const (
+	ContainerRuntimeDocker     = "docker"
+	ContainerRuntimeContainerd = "containerd"
+	ContainerRuntimeCRIO       = "cri-o"
+	timeout                    = 10
+	maxRows                    = 999
+
+	K8sDistroAKSAzureLinux           = "aks-AzureLinux"
+	K8sDistroAKSUbuntu               = "aks-Ubuntu"
+	K8sDistroEKSAmazonLinux          = "eks-AmazonLinux"
+	K8sDistroARO                     = "aro"
+	K8sDistroMinikubeGH              = "minikube-github"
+	securityProfileOperatorNamespace = "security-profiles-operator"
+	topTimeoutInSeconds              = 10
+)
+
+var (
+	supportedK8sDistros = []string{K8sDistroAKSAzureLinux, K8sDistroAKSUbuntu, K8sDistroARO, K8sDistroMinikubeGH}
+	cleaningUp          = uint32(0)
+
+	containerRuntime string
+	isDockerRuntime  bool
+	isCrioRuntime    bool
+)
+
+var (
+	integration    = flag.Bool("integration", false, "run integration tests")
+	dnsTesterImage = flag.String("dnstester-image", "ghcr.io/inspektor-gadget/dnstester:latest", "dnstester container image")
+	testComponent  = flag.String("test-component", "", "run tests for specific component")
+
+	image          = flag.String("image", "", "gadget container image")
+	doNotDeployIG  = flag.Bool("no-deploy-ig", false, "don't deploy Inspektor Gadget")
+	doNotDeploySPO = flag.Bool("no-deploy-spo", false, "don't deploy the Security Profiles Operator (SPO)")
+	k8sDistro      = flag.String("k8s-distro", "", "allows to skip tests that are not supported on a given Kubernetes distribution")
+	k8sArch        = flag.String("k8s-arch", "amd64", "allows to skip tests that are not supported on a given CPU architecture")
+)
+
+func normalizeCommonData(e *eventtypes.CommonData, ns string, cn string) {
```

Comment:
I see cn is only passed in integration/all/trace_open_test.go. I think we can remove this argument and handle the special case there (in integration/all/trace_open_test.go).